### ISSUE TITLE
Discarding changes on an file open twice in split buffers crashes Femto-Emacs

### ISSUE THREAD

**Issue by siraben** [FemtoEmacs/Femto-Emacs#10]
Reproducing the following steps produces a segfault
1. Open a file in Femto-Emacs (e.g. main.c)
2. Split the buffer with `C-x 2`, the file should now be open in two buffers
3. Make a change
4. Kill the buffer with `C-x k` and answering  `y`
5. Move to the other buffer with `C-x o`
6. `C-x k` and answering `y` produces the segfault.

What's interesting is that doing step 6 but answering `n` does _not_ crash Femto-Emacs.

**Comment by hughbarney** [comment]
Agreed this is a bug.  fix is fairly simple.  What is happening is that the code is not checking if any other window has a reference to the buffer that is being killed in the current window.  It is  probably about a 10 line fix.  Will fix in the Femto repository when I get time.  https://github.com/hughbarney/femto
